### PR TITLE
Switch testsuite to use pure Kubernetes APIs

### PR DESCRIPTION
**Pull request by galderz**
And with that, be able to run testsuite on Minikube. To test:

```bash
make minikube-config
make minikube-start
make minikube-test
```

Has been tested with latest Minikube (1.1.0).

I think we should embrace vanilla Kubernetes APIs, switch to Minikube completely and remove OpenShift references. Vanilla Kubernetes support is required to make it to operatorhub.io anyway.

Note: CI script would need updated, so CI runs for this PR will fail for sure.

### CODE REVIEW

**Comment by galderz** [comment]
I've created issue #50 for this.

**Comment by galderz** [comment]
@rigazilla This is no longer preview, it's ready to go in now that CI has been switched to minikube too. /cc @gustavonalle

**Comment by gustavonalle** [comment]
LGTM! One suggestion is to trigger a lot of builds on travis (like 10) to see if there aren't random failures, as k8s and openshift are complicated beasts :)

File: Makefile [hunk 1]
```diff
@@ -90,6 +90,9 @@ minikube-start:
 minikube-run-local: build
 	build/minikube/run-local.sh ${NAMESPACE}
 
+minikube-test: build
```
Comment (gustavonalle) [inline Makefile]: TIP: If add a comment on top of the minikube related tasks, when someone types ```make``` it shows a summary of available targets.

**Comment by galderz** [comment]
Moving this back to preview so that we can add testing on okd as well to CI. This will give us guarantees that we're not doing anything that is not supported in okd.

@gustavonalle If you have time, can you see if our okd set up can be updated?

**Comment by gustavonalle** [comment]
You mean, gustavonalle/oc-cluster-up?

**Comment by galderz** [comment]
@gustavonalle Yup

**Comment by gustavonalle** [comment]
I will take a look

**Comment by galderz** [comment]
@rigazilla This is ready to go in again. CI now runs with both Minikube and OpenShift and the jobs run concurrently to save up time. I've also completely removed the `okd` client code since we can now guarantee that vanilla Kubernetes based client code works well regardless of the target platform.

**Comment by galderz** [comment]
@rigazilla Can you integrate it then?

**Comment by galderz** [comment]
Ah yes, missing adding comments on top of make targets

**Comment by rigazilla** [comment]
merged, thanks @galderz!

**Pull request merged**